### PR TITLE
Enh/small improvements to standard name table

### LINKED ISSUE #405: Look and feel for standard name table
I would like to see the standard name table to be more user friendly with some CSS.

- In particular, as first stated by @sadielbartholomew in #130, the table lacks spacing & looks cluttered.
I would also add that links (the `<a>` tag) were used to show that elements can be clicked in the table, but it's not a best practice to use links if something is not actually a link. It's would be fairly simple to add some css to improve how interactive the table is.
- There is also a broken link to unidata website.
- Plus, when I did #218, I used some javascript that can easily be replaced by HTML5, which is often cleaner and easier to work with.

I already made a PR to fix these: #404.
The changes are displayed on a github page here: https://bzah.github.io/cf-convention.github.io/Data/cf-standard-names/current/build/cf-standard-name-table.html
The xsl file was updated and the html page was generated using the Makefile.

### PR DESCRIPTION
**Pull request by bzah**
This pull request add the following changes:
- Fixed minor typo in a README.
- Fixed Unidata link on the standard_name table page.
- Made use of HTML5 summary/details for "advanced search" helper box.
- Added some css to make the table prettier.
  - Remove arrows inviting for clicks in fqvor of a simple mouse :hover effect.
  - Made the mouse :hover style on the on the whole cell, not only the standard name text.
  -  Remove html tags making think the standard names are links which they are not.
  - Added small padding on cells
  - Remove table borders
  - Added a striped style for even rows of the table, it changes the background to a light pink for these rows.

This pull request would close #405 and **maybe** #130 too.

The changes from this pull request can be displayed on a github page here: https://bzah.github.io/cf-convention.github.io/Data/cf-standard-names/current/build/cf-standard-name-table.html

### CODE REVIEW

**Comment by JonathanGregory** [comment]
Thanks for doing this, Abel @bzah. Please could you open an _enhancement_ issue in this repo to describe it, and link this pull request?

**Comment by bzah** [comment]
@JonathanGregory yep sorry, here it is #405!

**Comment by bzah** [comment]
Hi @sadielbartholomew, are you interested in reviewing this PR ?
I think you did some similar changes for #130, so I would value your review greatly.
If you are not familiar with how the html page is created, you can mainly review the Data/cf-standard-names/current/xsl/html/cf-standard-name-table-1.3.xsl file. The html page is generated automatically from this `.xsl` document.

File: Data/cf-standard-names/current/xsl/html/cf-standard-name-table-1.3.xsl [hunk 1]
```diff
@@ -312,7 +333,7 @@ Refer to the <span class="link-external"><a href="http://cfconventions.org/Data/
...
...ventions.html#standard-name">Section 3.3 of the CF conventions</a></span> states: "Unless it is dimensionless, a variable with a standard_name attribute must have units which are physically equivalent (not necessarily identical) to the canonical units, possibly modified by an operation specified by either the standard name modifier ... or by the cell_methods attribute." Furthermore, <span class="link-external"><a href="http://cfconventions.org/cf-conventions/cf-conventions.html#_overview"> Section 1.3 of the CF conventions</a></span> states: "The values of the units attributes are character strings that are recognized by UNIDATA's Udunits package [UDUNITS], (with exceptions allowed as discussed in Section 3.1, “Units”)." For example, a variable with the standard name of "air_temperature" may have a units attribute of "degree_Celsius" because Celsius can be converted to Kelvin by Udunits. For the full range of supported units, refer to the <a href="https://docs.unidata.ucar.edu/udunits/current/#Database"> Udunits documentation</a>. Refer to the <a href="http://cfconventions.org/cf-conventions/cf-conventions.html"> CF conventions</a> for full details of the units attribute.<br/><br/>
```
Comment (bzah) [inline Data/cf-standard-names/current/xsl/html/cf-standard-name-table-1.3.xsl:336]: git seems a bit lost.
It's only a link update here, from `https://www.unidata.ucar.edu/software/udunits/udunits-current/doc/udunits/udunits2.html#Database"` to `https://docs.unidata.ucar.edu/udunits/current/#Database"`.

**Comment by larsbarring** [comment]
The CI checks throw an error, which also happens in PR #418. Does this prevent merging ?

**Comment by sadielbartholomew** [comment]
HI @bzah, sorry for not responding to your previous tag. Somehow this fell off my radar.

I can try to review by tomorrow if that works. If Alison can review before me, then go with her review, but I'll try to look shortly.

**Review by sadielbartholomew — COMMENTED** [review]
Brilliant stuff, thanks @bzah. Everything you promise in the opening comment summary has been achieved. I've eyeballed all of the code changes and they all look fine.

As for the change to the styling of the page, in my opinion you've definitely improved the look and feel, as well as the usability with your XSL style sheet additions. Towards #130, and to close it, I'd still prefer to add some further styling tweaks to make the page with the table much more consistent with the rest of the `cfconventions.org` site, in particular and at least changing the font to match (I believe this would be [Bootstrap's font specification](https://www.w3schools.com/bootstrap4/bootstrap_typography.asp)), but this could be done as part of another PR. However, if you'd like to include that here I am happy to consider #130 closed with this PR.

I also have a few minor comments, which I'd like you to consider before I provide a formal review approval, namely:
1. Now the arrow has been removed against each name, it might not be obvious to some that the names are clickable (assume they don't have much tech experience so don't know to, or end up, hovering over with their mouse). I think therefore we should be explicit and add a brief sentence to the top of the table to say something like 'Click on a name to show or hide its description and help text.'
2. Ideally the highlighting on the row of the name hovered over could cover all columns, not just the 'Standard Name' column, i.e. at the moment we have:

    scr_sn_table_style

**Review by sadielbartholomew — COMMENTED** [review]
I forgot to add, I am not sure we need to commit any changes to `Data/cf-standard-names/current/build/cf-standard-name-table.html`. Why have you included those? I can't actually see what they are since the `diff` is too large for the GitHub UI to display and I can't seem to fetch your branch - where does it live, since it doesn't get pulled down when I fetch your fork of the repo, not does it seem to be on the upstream repo?

**Comment by bzah** [comment]
@sadielbartholomew Many thanks for your review.

Regarding how the standard name table is different from the rest of CF:
Just to contextualize, most of the CF website is generated using markdown files, but the standard name table is not. Instead, it is a standalone html file which is generated from an xslx template using a xsltproc (with a makefile).
That's why there are inconsistencies with the rest of the website.
I have now did the same as you initially suggested and added the bootstrap.min.css to this page. That should fix the font issue and make other stuff like buttons more in the style of the rest of the CF website.

Regarding the highlights, I respectfully disagree. For now, only the cells within standard-name columns are clickable, if you click on another column nothing happen. I believe it is better to highlight only what is clickable, so I would suggest to leave this as is or to also make the whole row clickable. But, I'm not sure of the later because clicking only unrolls the text in the standard-name cell of the row. What do you think ? 

As for the html page, I generated it mainly to be able to publish it as a [github page here](https://bzah.github.io/cf-convention.github.io/Data/cf-standard-names/current/build/cf-standard-name-table.html): 
I don't mind reverting the html doc as it makes the changes unnecessarily big, and the best practices would suggest to leave generated files outside git anyway.
It's strange that you cannot see my branch, in any case you can see the html file here: https://github.com/bzah/cf-convention.github.io/tree/enh/small-improvements/Data/cf-standard-names/current/build 
And to get my branch on my fork, I would try something like:
```sh
git remote add bzah-fork https://github.com/bzah/cf-convention.github.io.git
git fetch bzah-fork
git checkout enh/small-improvements
```

**Comment by sadielbartholomew** [comment]
Thanks @bzah for your comments.

> That's why there are inconsistencies with the rest of the website.

I was aware of the context behind there being the stylistic differences, but my thinking was regardless it doesn't mean we can't bring things more in line with the rest of the website despite the under-the-hood difference in creation and format. Hence my comment.

> I have now did the same as you initially suggested and added the bootstrap.min.css to this page. That should fix the font issue and make other stuff like buttons more in the style of the rest of the CF website.

Perfect, let's a lot better I think, in line with my opening comment on #130, and means I am happy for this PR to resolve and close that one.

> Regarding the highlights, I respectfully disagree. For now, only the cells within standard-name columns are clickable, if you click on another column nothing happen. I believe it is better to highlight only what is clickable, so I would suggest to leave this as is or to also make the whole row clickable.

Fair enough. On the other hand I still like and prefer the alternative because the whole column is highlighted to indicate all of the relevant information, but overall it's a minor difference so not a blocker and you have your justifications for how you've done it, so that's fine by me to leave how it is. I imagine other people might prefer one or the other for similar reasons so either is good.

> I don't mind reverting the html doc as it makes the changes unnecessarily big, and the best practices would suggest to leave generated files outside git anyway.

Yes, please do revert that change. It bloats the PR and the commit history without much need to, I agree.

> It's strange that you cannot see my branch, in any case you can see the html file here

Indeed. I wonder if it is something to do with the `/` in the branch name, which I have not encountered before. I wonder if it needs escaping somehow. Either way, I am satisfied by what I have seen without the need to pull down the changes. IN this case there aren't e.g. tests I should run locally as part of the review, so it's OK.

****

Overall, once `Data/cf-standard-names/current/build/cf-standard-name-table.html` is removed from the PR, I am happy to approve and we can get this merged. So please do that when you get a chance. Thanks again.

**Comment by bzah** [comment]
I reset the html file to its original state.
I thinks it's ready to merge, thanks for the comments !

**Comment by larsbarring** [comment]
Many thanks @bzah for this brilliant contribution! And thanks @sadielbartholomew reviewing and comments. I will now merge this.

**Pull request merged**